### PR TITLE
Updated KNN calculations with formulas

### PR DESCRIPTION
**Pull request by keithhc2**
Added formulas for how to calculate Euclidian distance and cosine similarity, as well as scoring conversions for approximate k-NN and custom scoring

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/knn/index.md [hunk 1]
```diff
@@ -125,6 +125,34 @@ GET my-knn-index-1/_search
 }
 ```
 
+The formulas we use to perform calculations are as follows:
```
Comment (aetter) [inline docs/knn/index.md:128]: So I don't love the double : pattern here. How do you feel about removing this sentence and just doing the following?

Euclidian distance formula:

Cosine similarity formula:

File: docs/knn/index.md [hunk 2]
```diff
@@ -125,6 +125,34 @@ GET my-knn-index-1/_search
...
+Cosine similarity:
+<p>
+  \[ {A &middot; B \over \|A\| &middot; \|B\|} =
+
+  {\sum_{i=1}^n (A_i &middot; B_i) \over \sqrt{\sum_{i=1}^n A_i^2} &middot; \sqrt{\sum_{i=1}^n B_i^2}}\]
+ where \(\|A\|\) and \(\|B\|\) represent normalized vectors.
+
+</p>
+
+After calculations, we perform the following conversions to get the final Elasticsearch score. Note that these conversions are how we calculate approximate k-NN scores. For custom scoring conversions, see the [custom scoring](#Pre-filtering-with-custom-scoring) section.
```
Comment (aetter) [inline docs/knn/index.md:146]: Typically, we'd use "we" for "we, the humans who run this service" or "we, the maintainers of ODFE." In this case, the KNN plugin is performing the conversion, so we want to say something like, "After base calculations, the k-NN plugin performs the following conversions to get the final Elasticsearch relevance score. Note that these conversions are for approximate k-NN scores. For custom scoring conversions, see..."

File: docs/knn/index.md [hunk 3]
```diff
@@ -125,6 +125,34 @@ GET my-knn-index-1/_search
...
+
+[Euclidian Distance](https://github.com/opendistro-for-elasticsearch/k-NN/blob/0da03b29f1367b7f555e14b4ea4002626160bb35/src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNWeight.java#L113):
+```
+Elasticsearch score = 1/(1 + Euclidian distance)
```
Comment (aetter) [inline docs/knn/index.md:150]: I'd add a couple spaces here for readability and consistency, regardless of how it looks in code. `1 / (1 + Euclidian distance)` I think this pattern occurs four times.

File: docs/knn/index.md [hunk 4]
```diff
@@ -125,6 +125,34 @@ GET my-knn-index-1/_search
...
+  \[ {A &middot; B \over \|A\| &middot; \|B\|} =
+
+  {\sum_{i=1}^n (A_i &middot; B_i) \over \sqrt{\sum_{i=1}^n A_i^2} &middot; \sqrt{\sum_{i=1}^n B_i^2}}\]
+ where \(\|A\|\) and \(\|B\|\) represent normalized vectors.
+
+</p>
+
+After calculations, we perform the following conversions to get the final Elasticsearch score. Note that these conversions are how we calculate approximate k-NN scores. For custom scoring conversions, see the [custom scoring](#Pre-filtering-with-custom-scoring) section.
+
+[Euclidian Distance](https://github.com/opendistro-for-elasticsearch/k-NN/blob/0da03b29f1367b7f555e14b4ea4002626160bb35/src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNWeight.java#L113):
```
Comment (aetter) [inline docs/knn/index.md:148]: Same here, I think, as above. Something like...

Euclidian distance conversion:

Cosine similarity conversion:

File: docs/knn/index.md [hunk 5]
```diff
@@ -241,6 +269,18 @@ All parameters are required.
 - `query_value` is the point you want to find the nearest neighbors for. For the Euclidean and cosine similarity spaces, the value must be an array of floats that matches the dimension set in the field's mapping. For Hamming bit distance, this value can be either of type signed long or a base64-encoded string (for the long and binary field types, respectively).
 - `space_type` is either `l2`, `cosinesimil`, or `hammingbit`.
 
+After calculating distance, we perform the following conversions to return a score for your query.
+
+[Euclidian Distance](https://github.com/opendistro-for-elasticsearch/k-NN/blob/0da03b29f1367b7f555e14b4ea4002626160bb35/src/main/java/com/amazon/opendistroforelasticsearch/knn/plugin/script/KNNScoringSpace.java#L71):
```
Comment (aetter) [inline docs/knn/index.md:274]: Euclidian distance conversion:

File: docs/knn/index.md [hunk 6]
```diff
@@ -241,6 +269,18 @@ All parameters are required.
...
+```
+Elasticsearch score = 1/(1 + Euclidian distance)
+```
+[Cosine Similarity](https://github.com/opendistro-for-elasticsearch/k-NN/blob/3595be5b044205fbf5c02b2ecb68ff1df2a85b53/src/main/java/com/amazon/opendistroforelasticsearch/knn/plugin/script/KNNScoringSpace.java#L102):
```
Comment (aetter) [inline docs/knn/index.md:278]: Cosine similarity conversion

**Review by aetter — CHANGES_REQUESTED** [review]
Looks really good, just some minor nitpicks to clean up.

File: docs/knn/index.md (hunk 1 above)
Comment (keithhc2) [reply docs/knn/index.md:128]: That does look cleaner, so I'll go with your suggestion.

File: docs/knn/index.md (hunk 3 above)
Comment (keithhc2) [reply docs/knn/index.md:150]: I found three instances of this, and I've added spaces for readability, but please let me know if I need to edit anywhere else.

**Pull request merged**